packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -299,12 +299,13 @@ export const GridRootStyles = styled('div', {
       `]: {
       [`& .${c.columnSeparator}`]: {
         opacity: 0,
-        '@media (hover: none)': {
+      },
+      // Show resizable separators at all times on touch devices
+      '@media (hover: none)': {
+        [`& .${c['columnSeparator--resizable']}`]: {
           opacity: 1,
-          color: (t.vars || t).palette.primary.main,
```

Comment:
This was problematic - the conditions of the separator being the primary color are reduced on touch devices. Moved here: https://github.com/mui/mui-x/pull/14471/files#diff-e1db53193a627d36d1cd3f651a954ee1e32ecb93caa020446631d722bb1be75cR415-R422